### PR TITLE
Feature/smart crop

### PR DESCRIPTION
**Pull request by guillaumeogier**
This pull request include two smart crop to improve the original crop proposed by imagine. 
One is a crop by entropy and second is a crop by balanced.

thanks

### CODE REVIEW

**Comment by guillaumeogier** [comment]
If you want to see how it work or just see the difference between theses crops and the classical you can go to https://github.com/guillaumeogier/smart-crop, and watch the doc or the result of cropping

File: lib/Imagine/Effects/EffectsInterface.php [hunk 1]
```diff
@@ -77,4 +77,15 @@ public function sharpen();
      * @throws RuntimeException
      */
     public function blur($sigma);
+
```
Comment (Koc) [inline lib/Imagine/Effects/EffectsInterface.php]: file permissions here and below whould be reverted

**Comment by Koc** [comment]
@guillaumeogier can you also update documentation?

**Comment by burzum** [comment]
@Koc can you do and do another PR? I'm about to go to bed. I've missed the permissions. :( My bad.

**Comment by Koc** [comment]
I know nothing about this feature, so it would more logical that contributor will add docs about new things that he introduced

**Pull request merged**